tests/components/broadlink/test_config_flow.py

```diff
@@ -0,0 +1,527 @@
+"""Test the Broadlink config flow."""
+import errno
+import socket
+
+import broadlink.exceptions as blke
+import pytest
+
+from homeassistant import config_entries
+from homeassistant.components.broadlink.const import DOMAIN
+
+from . import pick_device
+
+from tests.async_mock import call, patch
+
+
+@pytest.fixture(autouse=True)
+def broadlink_setup_fixture():
+    """Mock broadlink entry setup."""
+    with patch(
+        "homeassistant.components.broadlink.async_setup_entry", return_value=True
+    ):
+        yield
+
+
+async def test_flow_user_works(hass):
+    """Test a config flow initiated by the user.
+
+    Best case scenario with no errors or locks.
+    """
+    device = pick_device(0)
+    mock_api = device.get_mock_api()
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "user"
+    assert result["errors"] == {}
+
+    with patch("broadlink.discover", return_value=[mock_api]) as mock_discover:
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {"host": device.host, "timeout": device.timeout},
+        )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "finish"
+    assert result["errors"] == {}
+
+    result = await hass.config_entries.flow.async_configure(
```

Comment:
We normally patch the integration `async_setup_entry` function for `create_entry` results to avoid setting up the integration.

Reply:
I'm doing it on line 20.